src/main/java/com/bravo/user/service/PaymentService.java

```diff
@@ -0,0 +1,42 @@
+package com.bravo.user.service;
+
+import com.bravo.user.dao.model.Payment;
+import com.bravo.user.dao.model.mapper.ResourceMapper;
+import com.bravo.user.dao.repository.PaymentRepository;
+import com.bravo.user.model.dto.PaymentDto;
+import com.bravo.user.utility.PageUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+@Service
+public class PaymentService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserService.class);
+
+    private final PaymentRepository paymentRepository;
+    private final ResourceMapper resourceMapper;
+
+    public PaymentService(PaymentRepository paymentRepository, ResourceMapper resourceMapper) {
+        this.paymentRepository = paymentRepository;
+        this.resourceMapper = resourceMapper;
+    }
+
+    public List<PaymentDto> retrieveByUserId(
+            final String userId,
+            final PageRequest pageRequest,
```

Comment:
With the values in pageRequest being able to be null.  Should this be optional?

Reply:
I prefer to not have optional arguments, also I was modeling this after the UserResource and was following the same coding style and design that was already established. If I was designing this I would have to separate methods one that expects a PageRequest, and one that does not.